src/Microsoft.ML.Core/Data/Repository.cs

```diff
@@ -525,7 +524,7 @@ public Entry OpenEntryOrNull(string dir, string name)
                     // Extract to a temporary file.
                     Directory.CreateDirectory(Path.GetDirectoryName(pathTemp));
                     entry.ExtractToFile(pathTemp);
-                    PathMap.Add(pathLower, pathTemp);
+                    PathMap.TryAdd(pathLower, pathTemp);
```

Comment:
To maintain similar to the previous behavior, we'll have to throw if the `TryAdd()` returns false.

The previous `Dictionary.Add()` throws an `ArgumentException` if the value being added already exists ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.add?view=netcore-3.0#exceptions)).

```suggestion
                    if (!PathMap.TryAdd(pathLower, pathTemp))
                        throw ExceptionContext.ExceptParam(nameof(name), "Duplicate entry: '{0}'", pathLower); 
```

Reply:
Updated to throw exception.